Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=f442cd86c1c86c5f44bc2cf23f89536f7e4cfe59
Commit:     f442cd86c1c86c5f44bc2cf23f89536f7e4cfe59
Parent:     2d3b8eea7f2fbafd5d779cc92f7aedbd1ef575e9
Author:     Albert Lee <albertcc@tw.ibm.com>
AuthorDate: Thu Nov 15 10:35:47 2007 +0900
Committer:  Tejun Heo <htejun@gmail.com>
CommitDate: Mon Nov 19 12:28:11 2007 +0900

    libata: use ATA_HORKAGE_STUCK_ERR for ATAPI tape drives
    
    Per Mark's comments, maybe all ATAPI tape drives need ATA_HORKAGE_STUCK_ERR.
    This patch applys ATA_HORKAGE_STUCK_ERR for all ATAPI tape drives.
    
    Signed-off-by: Albert Lee <albertcc@tw.ibm.com>
    Cc: Mark Lord <liml@rtr.ca>
    Signed-off-by: Tejun Heo <htejun@gmail.com>
---
 drivers/ata/libata-core.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- linux-2.6.23.noarch.orig/drivers/ata/libata-core.c
+++ linux-2.6.23.noarch/drivers/ata/libata-core.c
@@ -2020,8 +2020,10 @@ int ata_dev_configure(struct ata_device 
 	}
 
 	if ((dev->class == ATA_DEV_ATAPI) &&
-	    (atapi_command_packet_set(id) == TYPE_TAPE))
+	    (atapi_command_packet_set(id) == TYPE_TAPE)) {
 		dev->max_sectors = ATA_MAX_SECTORS_TAPE;
+		dev->horkage |= ATA_HORKAGE_STUCK_ERR;
+	}
 
 	if (dev->horkage & ATA_HORKAGE_MAX_SEC_128)
 		dev->max_sectors = min_t(unsigned int, ATA_MAX_SECTORS_128,
